lib/pipeline.js

```diff
@@ -135,6 +135,82 @@ Pipeline.prototype = {
   },
 
   /**
+   * Bind to a streams events, and re-emit them on the pipeline
+   * @param {Stream} stream
+   * @param {String|Array} events
+   * @returns {Pipeline}
+   * @chainable
+   */
+  bind( stream, events ) {
```

Comment:
> Hmmm, so it sounds like effectively these are actually methods on streams themselves, with the pipeline only being a parameter used in the bind call? Would it make sense to make them static methods on the Pipeline class

Nope, these do have an effect on the pipelines internals (event emission, handler context, ...), if you'd want a no-side-effects / static thing, you could just do `somestream.on( 'event', handler )`, instead of going through the pipeline.

> Hmmm? Doesn't the usage of unbindAll in splice mean that a removed stream automatically has all its events unbound?

Yes, but at that time they're already removed from the internal array – the reasoning there being, that if the pipeline is bound to a stream's `unpipe` event, or any other event that might result out of unpiping, those events wouldn't be emitted if the `.unbindAll()` were to happen before removal.